assisted-events-scrape/utils/anonymizer.py

```diff
@@ -13,12 +13,12 @@ def anonymize_infra_env(cls, infra_env: dict):
     @classmethod
     def _hash_user_name(cls, resource: dict):
         """mask user name with unique generated FNV-1a 128 bit id"""
-        if "user_name" not in resource:
+
+        if type(resource) is not dict or "user_name" not in resource:
```

Comment:
in that case, the signature of the methods for this class should be `resource: Any`, no?

Reply:
It should be dict, but somehow we get some input like this (see screenshot above). I can't understand exactly what it is, but this is one option (although it's very weird this is happening).
Unfortunately typing is not enough to prevent it.

Reply:
I agree that you cannot change in input data, but it makes the signature invalid

Reply:
also, the signature is a form of documentation, so it makes clear that the input data need to be handled carefully.

Reply:
Make sense, but the good solution here would be tracking down who in the world is returning a non-dict type :sweat: